federation/pkg/federation-controller/sync/controller.go

```diff
@@ -384,6 +391,50 @@ func (s *FederationSyncController) delete(obj pkgruntime.Object, kind string, na
 	return nil
 }
 
+// updateFedSpec updates the spec against the federated object using the adapters UpdateFedSpecHook of need be.
+func (s *FederationSyncController) updateFedSpec(fedObj pkgruntime.Object, hook federatedtypes.UpdateFedSpecHook) reconcileStatus {
```

Comment:
The intention of UpdateFedSpecHook (wrongly named hook function as named earlier, updated to FedUpdateStatus in the process of fixing the review comments), is to update the federated object status to reflect the status from all clusters. I think the confusion happened because of me using 'Spec' in place of 'Status' in the name (updated now). Apologies!
Additionally, keeping this method separate (from Schedule()) originally was the thought that there might be some controllers where status update into federation object might not be needed. The whole work can as well be done in one method, because both (schedule and determining collective cluster object status) which need to iterate over the cluster objects, might better be done in one pass. (Let me know if its more desirable, especially following your suggested principle of YAGNI.. :) I will be happy to update it).